src/uucore/src/lib/features/backup_control.rs

```diff
@@ -416,25 +416,34 @@ pub fn get_backup_path(
 }
 
 fn simple_backup_path(path: &Path, suffix: &str) -> PathBuf {
-    let mut p = path.to_string_lossy().into_owned();
-    p.push_str(suffix);
-    PathBuf::from(p)
+    match path.file_name() {
+        Some(file_name) => {
+            let mut file_name = file_name.to_os_string();
+            file_name.push(suffix);
+            path.with_file_name(file_name)
+        }
+        None => path.with_file_name(suffix),
+    }
```

Comment:
can be simplify this way:

```suggestion
    let file_name = path.file_name().unwrap_or_default().to_os_string();
    path.with_file_name(file_name.to_string_lossy().to_string() + suffix)
```

Reply:
sure, sorry :)